### PR TITLE
Create endpoint to handle the reporting of bad veteran representative contact data

### PR DESCRIPTION
**Pull request by holdenhinkle**
In regards to https://github.com/department-of-veterans-affairs/va.gov-team/issues/72661

## Summary
This PR creates a new endpoint for users to flag bad contact info for reps (email, phone, address), or submit a note about an issue (other).

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/72661

## Testing done
- [x] *New code is covered by unit tests*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by cosu419 — DISMISSED** [review]

File: modules/veteran/app/controllers/veteran/v0/flag_accredited_representatives_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Veteran
+  module V0
+    class FlagAccreditedRepresentativesController < ApplicationController
+      service_tag 'lighthouse-veteran'
+      skip_before_action :verify_authenticity_token
+      skip_before_action :authenticate
+      before_action :feature_enabled
```
Comment (jvcAdHoc) [inline modules/veteran/app/controllers/veteran/v0/flag_accredited_representatives_controller.rb:9]: I think we will want some form of authentication. Unlike the find a rep controller, there are in-system and user facing consequences to using this endpoint.

File: modules/veteran/app/models/veteran/flagged_veteran_representative_contact_data.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Veteran
+  class FlaggedVeteranRepresentativeContactData < ApplicationRecord
+    self.table_name = 'flagged_veteran_representative_contact_data'
+
+    attribute :flag_type, :flag_type
+    validates :ip_address, :representative_id, :flag_type, :flagged_value, presence: true
+    validates :flag_type, inclusion: { in: ActiveRecord::Type.lookup(:flag_type).valid_types, message: 'Invalid flag type: must be phone, email, address, or other' } # rubocop:disable Layout/LineLength,Rails/I18nLocaleTexts
```
Comment (jvcAdHoc) [inline modules/veteran/app/models/veteran/flagged_veteran_representative_contact_data.rb:9]: I think a cleaner implementation is defining `flag_type` as a string defined enum on this model

File: modules/veteran/app/controllers/veteran/v0/flag_accredited_representatives_controller.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+              flag_data.permit(:flag_type, :flagged_value).merge(
+                ip_address: request.remote_ip,
+                representative_id: params[:representative_id]
+              )
+            )
+          end
+
+          if flags.all?(&:valid?)
+            flags.each(&:save!)
+            render json: flags, status: :created
```
Comment (jvcAdHoc) [inline modules/veteran/app/controllers/veteran/v0/flag_accredited_representatives_controller.rb:24]: I think the render should be outside of the transaction block

File: modules/veteran/spec/requests/v0/flag_accredited_representatives_spec.rb
Comment (jvcAdHoc) [inline modules/veteran/spec/requests/v0/flag_accredited_representatives_spec.rb:1]: The tests don't cover the expected json output: what do error messages look like and what the serialized flags look like

File: modules/veteran/app/controllers/veteran/v0/flag_accredited_representatives_controller.rb (hunk 3 above)
Comment (holdenhinkle) [reply modules/veteran/app/controllers/veteran/v0/flag_accredited_representatives_controller.rb:18]: TY. Fixed.

File: modules/veteran/app/models/veteran/flagged_veteran_representative_contact_data.rb (hunk 2 above)
Comment (holdenhinkle) [reply modules/veteran/app/models/veteran/flagged_veteran_representative_contact_data.rb]: TY. I got it working.

**Comment by dbrown-va** [comment]
@holdenhinkle , are these features solely for VA.gov front-end features?

I see this and a couple other recent PR's in the `modules/veteran` directory which was originally tagged as a Lighthouse endpoint for looking up Veteran representatives as part of Claims API and other Lighthouse API functionality.

Concerned about the location and intermingling with VA.gov specific features in the module.

File: modules/veteran/app/controllers/veteran/v0/flag_accredited_representatives_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          render json: { errors: flags.map(&:errors).reject(&:empty?) }, status: :unprocessable_entity
+        end
+      rescue ArgumentError => e
+        render json: { errors: { flag_type: [e.message] } }, status: :unprocessable_entity
+      end
+
+      private
+
+      def feature_enabled
+        routing_error unless Flipper.enabled?(:find_a_representative_enable_api)
```
[the remark this answers is not in the document]
Comment (holdenhinkle) [reply modules/veteran/app/controllers/veteran/v0/flag_accredited_representatives_controller.rb]: Updated

File: modules/veteran/spec/requests/v0/flag_accredited_representatives_spec.rb
Comment (holdenhinkle) [reply modules/veteran/spec/requests/v0/flag_accredited_representatives_spec.rb:1]: Done.

**Comment by holdenhinkle** [comment]
> @holdenhinkle , are these features solely for VA.gov front-end features?
> 
> I see this and a couple other recent PR's in the `modules/veteran` directory which was originally tagged as a Lighthouse endpoint for looking up Veteran representatives as part of Claims API and other Lighthouse API functionality.
> 
> Concerned about the location and intermingling with VA.gov specific features in the module.

@dbrown-va We're planning on moving all of our code out of the veteran module into our own module real soon.

**Review by cosu419 — APPROVED** [review]
LGTM. Approved!

**Comment by dbrown-va** [comment]
> > @holdenhinkle , are these features solely for VA.gov front-end features?
> > I see this and a couple other recent PR's in the `modules/veteran` directory which was originally tagged as a Lighthouse endpoint for looking up Veteran representatives as part of Claims API and other Lighthouse API functionality.
> > Concerned about the location and intermingling with VA.gov specific features in the module.
> 
> @dbrown-va We're planning on moving all of our code out of the veteran module into our own module real soon.

Can we link a ticket here for tracking?  I see several other endpoints already added within this module and want to ensure this separation has definitive end date.

Lighthouse functionality will be extracted to separate DI deployed containers in the future and want to minimize that complexity.

**Comment by holdenhinkle** [comment]
@dbrown-va 
Sure. 

Here's a ticket about moving to our own module - https://app.zenhub.com/workspaces/accredited-representation-management-team-64d0dc51d3e8f4788ac6ef96/issues/gh/department-of-veterans-affairs/va.gov-team/71814

Here's a ticket I just create to move the code to our own module - https://app.zenhub.com/workspaces/accredited-representation-management-team-64d0dc51d3e8f4788ac6ef96/issues/gh/department-of-veterans-affairs/va.gov-team/75019

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -187,6 +187,16 @@
     ActionController::Base.allow_forgery_protection = @original_allow_forgery_protection
   end
 
+  # Disable CSRF protection for FlagAccreditedRepresentativesController specs in the Veteran module
+  config.before(:each, csrf: false) do
+    ActionController::Base.allow_forgery_protection = false
+  end
+
+  # Enable CSRF protection for FlagAccreditedRepresentativesController specs in the Veteran module
+  config.after(:each, csrf: false) do
```
Comment (ericboehs) [inline spec/spec_helper.rb:196]: Should this be csrf: true?

Comment (ericboehs) [reply spec/spec_helper.rb:196]: Hmm. I think I'm confused. Haha.

**Review by ericboehs — APPROVED** [review]

**Pull request merged**